src/unit_tests/wazuh_db/test_wdb_global_helpers.c

```diff
@@ -3635,12 +3535,377 @@ void test_wdb_parse_chunk_to_int_err(void **state) {
+    will_return(__wrap_cJSON_CreateObject, j_data_in);
+    will_return(__wrap_cJSON_AddStringToObject, 1);
+    expect_string(__wrap_cJSON_AddStringToObject, name, "mode");
+    expect_string(__wrap_cJSON_AddStringToObject, string, "override");
+    will_return(__wrap_cJSON_AddStringToObject, 1);
+    expect_string(__wrap_cJSON_AddStringToObject, name, "sync_status");
+    expect_string(__wrap_cJSON_AddStringToObject, string, "synced");
+    will_return(__wrap_cJSON_CreateObject, j_agent_info);
+    expect_function_call(__wrap_cJSON_AddItemToArray);
+    will_return(__wrap_cJSON_AddItemToArray, true);
+    expect_string(__wrap_cJSON_AddNumberToObject, name, "id");
+    expect_value(__wrap_cJSON_AddNumberToObject, number, 1);
+    will_return_always(__wrap_cJSON_AddNumberToObject, 1);
+
+    // Json array items loop
+    expect_string(__wrap_cJSON_CreateString, string, "default");
+    will_return(__wrap_cJSON_CreateString, 1);
+    expect_function_call(__wrap_cJSON_AddItemToArray);
+    will_return(__wrap_cJSON_AddItemToArray, true);
+    expect_string(__wrap_cJSON_CreateString, string, "Group1");
+    will_return(__wrap_cJSON_CreateString, 1);
+    expect_function_call(__wrap_cJSON_AddItemToArray);
+    will_return(__wrap_cJSON_AddItemToArray, true);
+    expect_string(__wrap_cJSON_CreateString, string, "Group2");
+    will_return(__wrap_cJSON_CreateString, 1);
+    expect_function_call(__wrap_cJSON_AddItemToArray);
+    will_return(__wrap_cJSON_AddItemToArray, true);
+
+    // Printing JSON
+    will_return(__wrap_cJSON_PrintUnformatted, data_in_str);
+    expect_function_call(__wrap_cJSON_Delete);
+
+    // Calling Wazuh DB
+    expect_value(__wrap_wdbc_query_ex, *sock, socket);
+    expect_string(__wrap_wdbc_query_ex, query, query_str);
+    expect_value(__wrap_wdbc_query_ex, len, WDBOUTPUT_SIZE);
+    will_return(__wrap_wdbc_query_ex, response);
+    will_return(__wrap_wdbc_query_ex, OS_SUCCESS);
+
+    // Parsing Wazuh DB result
+    expect_any(__wrap_wdbc_parse_result, result);
+    will_return(__wrap_wdbc_parse_result, WDBC_ERROR);
+
+    // Debug message
+    expect_string(__wrap__mdebug1, formatted_msg, "Global DB Error reported in the result of the query");
+
+    res = wdb_set_agent_groups(id, groups_array, mode, sync_status, &socket);
+
+    assert_int_equal(OS_INVALID,res);
+
+    free_strarray(groups_array);
+    os_free(mode);
+    os_free(sync_status);
+    os_free(query_str);
+    os_free(response);
+    __real_cJSON_Delete(j_data_in);
+    __real_cJSON_Delete(j_agent_info);
+}
+
+void test_wdb_set_agent_groups_success(void **state) {
+    char** groups_array = NULL;
+    char* mode = NULL;
+    char* sync_status = NULL;
+    char* query_str = NULL;
+    char* data_in_str = NULL;
+    char* response = NULL;
+    int id = 1;
+    int socket = -1;
+    int res;
+    cJSON* j_data_in = __real_cJSON_CreateObject();
+    cJSON* j_agent_info = __real_cJSON_CreateObject();
+
+    os_calloc(4, sizeof(char *), groups_array);
```

Comment:
same recommendations as above.

Additionally the setup and tear-down  code for all this last three functions is basically the same because we are trying to get to the `wdbc_query_ex` function call. It would be cleaner/better to just wrap them in a setup/teardown functions pair.